sdk/core/azure-core/inc/azure/core/match_conditions.hpp

```diff
@@ -20,6 +20,10 @@ namespace Azure {
  */
 struct MatchConditions
 {
+  /**
+   * @brief Destructs the MatchConditions.
```

Comment:
```suggestion
   * @brief Destructs the `MatchConditions`.
```

Please, review other places her yourself and backtick all the places where we don't want to self-refrence, but need to use the name of self class.